spec/spec_helper.rb

```diff
@@ -153,6 +156,22 @@ module PuppetSpec
     Puppet::Test::TestHelper.before_each_test()
   end
 
+  # If an example doesn't use VCR, then disable VCR entirely. Otherwise, webmock does not
+  # prevent unexpected connections from being attempted, most likely due to the way that
+  # we start connections explicitly using Net::HTTP#start, prior to the request. Although
+  # this is described in https://github.com/bblimke/webmock#connecting-on-nethttpstart,
+  # setting `WebMock.allow_net_connect!(net_http_connect_on_start: true)` doesn't help.
+  config.around :each do |example|
+    if example.metadata[:vcr]
+      example.run
+    else
+      VCR.turned_off do
+        stub_request(:get, "http://metadata/computeMetadata/v1beta1/?alt=json&recursive=true")
```

Comment:
Looks like GCE metadata service is changing to `http://metadata.google.internal/computeMetadata/v1/`, see FACT-2018, so I should make this more general

Reply:
actually that's not necessary since we won't be updating facter 2.x to use the new endpoint.